src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -1273,9 +1291,155 @@ export default {
 				this.showCheckout();
 			}
 		},
+		amountLeftOnLoan(loan) {
+			const loanFundraisingInfo = loan?.loanFundraisingInfo ?? { fundedAmount: 0, reservedAmount: 0 };
+			const { fundedAmount, reservedAmount } = loanFundraisingInfo;
+			return numeral(loan.loanAmount).subtract(fundedAmount).subtract(reservedAmount).value();
+		},
+		allocateLeftoverCredits(payload) {
+			const newPayload = payload;
+			let loanIdx = newPayload.loanIdx ? newPayload.loanIdx : 0;
+			// Allocate leftover credits to a loan that isn't already in the basket
+			// and make sure it needs at least the leftover amount we want to allocate
+			while (
+				this.itemsInBasket.indexOf(this.availableLoans.values[loanIdx].id) > -1
+				|| newPayload.lendAmount > this.amountLeftOnLoan(this.availableLoans.values[loanIdx])
+			) {
+				loanIdx += 1;
+				if (loanIdx === this.availableLoans.length - 1) {
+					break;
+				}
+			}
+			const loanId = this.availableLoans.values[loanIdx].id;
+			setLendAmount({
+				amount: newPayload.lendAmount,
+				apollo: this.apollo,
+				loanId
+			}).then(() => {
+				console.log(`Successfully added loan with id ${loanId} to basket`);
+				this.$emit('add-to-basket', { loanId, success: true });
+				this.leftoverCreditAllocationLoanId = loanId;
+				this.updateBasketState();
+			}).catch(e => {
+				console.log(`Failed to add loan with id ${loanId} to basket`);
```

Comment:
Do we want to use `logFormatter` here and add an error log?

Reply:
done